### ISSUE TITLE
RNP should use repgp interface

### ISSUE THREAD

**Issue by flowher** [riboseinc/rnp#523]
## Description

I think original idea was to introduce interface between ``rnp`` and ``repgp``. Then ``rnp`` (``rnp/lib/rnp.c``) should use this interface for all PGP data packets processing. So, to be done here is:

1. ``repgp`` should be using stream data processing
2. ``repgp`` should implement read and write (currently API offers only parsing API - so read) 
3. ``rnp.c`` should use API from ``repgp``

Thoughts?

**Comment by ronaldtse** [comment]
@flowher sorry for the late reply. By "PGP data processing", do you mean that `repgp` should handle the reading and writing of OpenPGP packets?

If `repgp` can do all that, what's left of the use of `rnp`? `rnp` provides the data to write OpenPGP packets through `repgp`?

**Comment by flowher** [comment]
Yes, that would be my suggestion. The ``rnp`` would be just a light wrapper around ``librepgp`` - basically implementation of the ``main()`` function, attribute parsing and everything else which is needed for ``rnp`` but not actual parsing/writing. 
The idea comes from the fact I don't see why we would need to differentiate between parsing and serializing. Actual implementation (location of files) for stream writing is already in ``librepgp`` anyway.
I'm interested in opinions.

**Comment by ronaldtse** [comment]
The other part of rnp other than the OpenPGP packet format is the keystore stuff. I do agree that reading and writing of the packet format should be done in the same library. 

What are your views, @dewyatt @ni4 @randombit?
